### ISSUE TITLE
Prefer RSA-AES128-GCM over RSA-AES256-CBC

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6906]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)| @hynek reported|
|-|-|
|Trac ID|trac#6906|
|Type|enhancement|
|Created|2014-01-17 14:20:45Z|
|Branch|https://github.com/twisted/twisted/tree/prefer-gcm-over-cbc-6906|

As Alex pointed out, until we support PFS (DH/ECDHE) key exchange, the current cipher string favors AES256-CBC over AES128-GCM. Although the former sounds more impressive, the latter is more modern, more secure, and faster.

The solution is to simply split up `RSA+AES` into `RSA+AESGCM:RSA+AES` so both kinds of GCM are preferred over CBC.

<details><summary>Searchable metadata</summary>

```
trac-id__6906 6906
type__enhancement enhancement
reporter__hynek hynek
priority__normal normal
milestone__ 
branch__branches_prefer_gcm_over_cbc_6906 branches/prefer-gcm-over-cbc-6906
branch_author__hynek hynek
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1389968445000000 1389968445000000
changetime__1389971188000000 1389971188000000
version__None None
owner__hynek hynek

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_1"></a>|@hynek commented|
|-|-|

(In [41306]) Branching to prefer-gcm-over-cbc-6906.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_2"></a>|@hynek commented|
|-|-|

Additionally to splitting AES, I’ve also added a more verbose comment about the intent behind the string to make it more comprehensible.

The bots are green, pyflakes and Win7 are unrelated.

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)|@hynek removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dstufft's avatar" src="https://avatars.githubusercontent.com/u/145979?s=50" width="50" height="50">](https://github.com/dstufft)|@dstufft set owner to @dstufft<br>@dstufft set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="dstufft's avatar" src="https://avatars.githubusercontent.com/u/145979?s=50" width="50" height="50">](https://github.com/dstufft)<a name="note_5"></a>|@dstufft set owner to @hynek<br>@dstufft set status to `new`|
|-|-|

LGTM

**Comment by twisted-trac** [comment]
|[<img alt="hynek's avatar" src="https://avatars.githubusercontent.com/u/41240?s=50" width="50" height="50">](https://github.com/hynek)<a name="note_6"></a>|@hynek set status to `closed`|
|-|-|

(In [41308]) Merge prefer-gcm-over-cbc-6906

Author: hynek
Reviewer: dstufft
Fixes: [#6906](https://github.com/twisted/twisted/issues/6906)

Prefer any kind of AES-GCM over any kind of AES-CBC.
Add intent behind cipher string.
